erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -203,14 +252,15 @@ def auto_fetch_serial_number(
 	for_doctype: Optional[str] = None,
 	exclude_sr_nos=None,
 ) -> List[str]:
+	"""NOTE: Unused but kept for backward compatibility."""
 
 	filters = frappe._dict({"item_code": item_code, "warehouse": warehouse})
 
 	if exclude_sr_nos is None:
 		exclude_sr_nos = []
 	else:
 		exclude_sr_nos = safe_json_loads(exclude_sr_nos)
-		exclude_sr_nos = get_serial_nos(clean_serial_no_string("\n".join(exclude_sr_nos)))
+		exclude_sr_nos = get_serial_nos(clean_serial_no_string("\n".join(exclude_sr_nos), item_code))
```

Comment:
`get_serial_nos` is mainly used to handle legacy data where the serial no text field is used.
Anyway, it will return the serial no back as is if it cannot be found to keep it consistent with older behaviour.
The retrieval of serial no name is added as a defensive approach.